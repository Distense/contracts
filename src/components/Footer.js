import React from 'react'
import { Link } from 'react-router-dom'
import EthereumAuthenticated from './common/EthereumAuthenticated'
// import GasEstimate from './common/GasEstimate'
import NumberDIDOwned from './common/NumberDIDOwned'
import HasWeb3 from './common/HasWeb3'

import { Container, Menu, Segment } from 'semantic-ui-react'

import Status from './Status'

export default () => (
  <Segment vertical style={{ margin: '3em 0em 0em', padding: '5em 0em' }}>
    <Menu className="inconsolata" borderless fixed="bottom">
      <Container>
        <Menu.Item>Â© {new Date().getFullYear()} Distense</Menu.Item>
        <Menu.Item
          style={{ textDecoration: 'underline' }}
          color="blue"
          link
          as={Link}
          position="right"
          to="/getstarted"
        >
          Get Started
        </Menu.Item>
        <Menu.Item
          style={{ textDecoration: 'underline' }}
          color="blue"
          link
          as={Link}
          to="/FAQ"
        >
          FAQ
        </Menu.Item>
        <Menu.Item
          style={{ textDecoration: 'underline' }}
          color="blue"
          link
          as={Link}
          to="/parameters"
        >
          Parameters
        </Menu.Item>
        <Menu.Item
          style={{ textDecoration: 'underline' }}
          color="blue"
          link
          as={Link}
          to="/events"
        >
          Event Feed
        </Menu.Item>
        <Status />
        <NumberDIDOwned />
        {/*<GasEstimate />*/}
        <HasWeb3 />
        <EthereumAuthenticated />
      </Container>
    </Menu>
  </Segment>
)
