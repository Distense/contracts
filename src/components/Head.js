import React from 'react'
import { Helmet } from 'react-helmet'

export default ({ title, children }) => (
  <Helmet>
    <title>{title ? `${title} - Distense` : 'Distense'}</title>
    {children}
  </Helmet>
)
