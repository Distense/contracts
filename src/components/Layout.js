import React from 'react'
import { Link } from 'react-router-dom'
import { Container, Menu, Segment } from 'semantic-ui-react'

import EthereumAuthenticated from './common/EthereumAuthenticated'
// import GasEstimate from './common/GasEstimate'
import NumberDIDOwned from './common/NumberDIDOwned'
import HasWeb3 from './common/HasWeb3'
import TotalDID from './common/TotalDID'

import Status from './Status'

export default ({ children, title }) => (
  <div>
    <Segment
      inverted
      textAlign="center"
      style={{
        padding: '.8em 0em'
      }}
      vertical
    >
      <Menu
        borderless
        fixed="top"
        className="inconsolata"
        inverted
        size="large"
      >
        <Container textAlign="center">
          <Menu.Item to="/" as={Link} position="left">
            Distense
          </Menu.Item>
          <Menu.Item to="/tasks/add" as={Link}>
            Propose
          </Menu.Item>
          <Menu.Item to="/tasks" as={Link}>
            View
          </Menu.Item>
          <Menu.Item to="/pullrequests/add" as={Link}>
            Submit
          </Menu.Item>
          <Menu.Item to="/pullrequests" as={Link}>
            Approve
          </Menu.Item>
          <Menu.Item to="/exchange" as={Link}>
            Exchange
          </Menu.Item>
          <TotalDID />
        </Container>
      </Menu>
    </Segment>

    <Container style={{ marginTop: '4em' }}>{children}</Container>

    <Segment vertical style={{ margin: '3em 0em 0em', padding: '5em 0em' }}>
      <Menu className="inconsolata" borderless fixed="bottom">
        <Container>
          <Menu.Item>Â© {new Date().getFullYear()} Distense</Menu.Item>
          <Menu.Item
            style={{ textDecoration: 'underline' }}
            color="blue"
            link
            as={Link}
            position="right"
            to="/FAQ"
          >
            FAQ
          </Menu.Item>
          <Menu.Item
            style={{ textDecoration: 'underline' }}
            color="blue"
            link
            as={Link}
            to="/parameters"
          >
            Parameters
          </Menu.Item>
          <Menu.Item
            style={{ textDecoration: 'underline' }}
            color="blue"
            link
            as={Link}
            to="/events"
          >
            Event Feed
          </Menu.Item>
          <Status />
          <NumberDIDOwned />
          {/*<GasEstimate />*/}
          <HasWeb3 />
          <EthereumAuthenticated />
        </Container>
      </Menu>
    </Segment>
    {/*language=CSS*/}
    <style global jsx>{`
      .inconsolata {
        font-family: 'Inconsolata', sans-serif !important;
      }
    `}</style>
  </div>
)
