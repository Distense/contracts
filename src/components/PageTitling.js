import React from 'react'
import { Grid, Header } from 'semantic-ui-react'

export default ({ title, subtitle }) => {
  return (
    <Grid.Column width={16}>
      <Header as="h2">{title}</Header>
      <p style={{ fontSize: '1.3rem' }}>{subtitle}</p>
    </Grid.Column>
  )
}
