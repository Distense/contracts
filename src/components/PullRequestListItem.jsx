import React, { Component } from 'react'
import { connect } from 'react-redux'
import { Button, Table } from 'semantic-ui-react'
import { PAID } from '../constants/rewardStatuses'

import { approvePullRequest } from '../actions/pullRequests'

import Tags from '../components/common/Tags'

class PullRequestsListItem extends Component {
  constructor(props) {
    super(props)
    this.state = {
      pr: this.props.pullRequest,
      approveText:
        this.props.pullRequest.status === 'PAID' ? 'Approved' : 'Approve ',
      approveButtonDisabled: false
    }
    this.onClickApprove = this.onClickApprove.bind(this)
  }

  onClickApprove = async event => {
    event.preventDefault()

    this.setState({
      approveText: 'Approving PR',
      approveButtonDisabled: true
    })

    this.props.approvePullRequest(this.state.pr._id)
  }

  render() {
    const { approveText, pr } = this.state

    let approveButtonColor
    let approveButtonDisabled
    if (pr.status === PAID) {
      approveButtonColor = 'black'
      approveButtonDisabled = true
    } else {
      approveButtonColor = 'green'
      approveButtonDisabled = false
    }

    return (
      <Table.Row key={pr._id}>
        <Table.Cell>{pr.taskTitle}</Table.Cell>
        <Table.Cell>
          <Tags tags={pr.tags} />
        </Table.Cell>
        <Table.Cell>{pr.rewardStatus}</Table.Cell>
        <Table.Cell>{pr.taskReward}</Table.Cell>
        <Table.Cell>{pr.pctDIDApproved}</Table.Cell>
        <Table.Cell>
          <Button
            basic
            color={approveButtonColor}
            compact={true}
            floated="right"
            fluid={true}
            disabled={approveButtonDisabled}
            size="mini"
            pr={pr}
            onClick={e => this.onClickApprove(e, pr)}
          >
            {approveText}
          </Button>
        </Table.Cell>
        <Table.Cell>
          <Button
            color="green"
            compact={true}
            floated="right"
            fluid={true}
            size="mini"
            target="_blank"
            href={`${pr.url}`}
          >
            Review PR
          </Button>
        </Table.Cell>
      </Table.Row>
    )
  }
}

const mapStateToProps = () => ({})

const mapDispatchToProps = dispatch => ({
  approvePullRequest: task => dispatch(approvePullRequest(task))
})

export default connect(mapStateToProps, mapDispatchToProps)(
  PullRequestsListItem
)
