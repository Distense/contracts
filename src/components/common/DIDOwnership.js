import React, { Component } from 'react'
import { Menu } from 'semantic-ui-react'
import { connect } from 'react-redux'

import { fetchTotalSupplyDID } from '../../actions/status'
import { getTotalSupplyDID } from '../../reducers/status'

export class DIDOwnership extends Component {
  componentDidMount() {
    this.props.fetchTotalSupplyDID()
  }

  componentWillUpdate(nextProps) {
    return (
      this.props.numDID !== nextProps.numDID ||
      this.props.totalSupplyDid !== nextProps.totalSupplyDid
    )
  }

  calculatePercentageOwned() {
    const { numDID, totalSupplyDid } = this.props

    let pctDID = (numDID / totalSupplyDid * 100).toFixed(2)

    pctDID = isNaN(pctDID) ? '0' : pctDID
    pctDID = pctDID === '0.00' ? '0' : pctDID
    pctDID = pctDID === '100.00' ? '100' : pctDID

    return pctDID
  }

  render() {
    const { numDID } = this.props
    const pctDID = this.calculatePercentageOwned()

    return (
      <Menu.Item
        className="footer-item"
        title="The number of and percentage of total DID the current account owns"
      >
        DID owned: {numDID} ({pctDID}%)
      </Menu.Item>
    )
  }
}

const mapStateToProps = state => ({
  numDID: state.user.user.numDID,
  totalSupplyDid: getTotalSupplyDID(state)
})

const mapDispatchToProps = dispatch => ({
  fetchTotalSupplyDID: () => dispatch(fetchTotalSupplyDID())
})

export default connect(mapStateToProps, mapDispatchToProps)(DIDOwnership)
