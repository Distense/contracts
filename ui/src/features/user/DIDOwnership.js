import React, { Component } from 'react'
import { Menu } from 'semantic-ui-react'
import { connect } from 'react-redux'

import { fetchTotalSupplyDID } from '../distense/actions'
import { getTotalSupplyDID } from '../distense/reducers'
import { calcPctDIDOwned } from './actions'

export class DIDOwnership extends Component {
  componentDidMount() {
    this.props.fetchTotalSupplyDID()
  }

  UNSAFE_componentWillUpdate(nextProps) {
    return (
      this.props.numDID !== nextProps.numDID ||
      this.props.totalSupplyDid !== nextProps.totalSupplyDid
    )
  }

  render() {
    const { numDID } = this.props
    const pctDID = calcPctDIDOwned(numDID)

    return (
      <Menu.Item
        className="footer-item"
        title="The number of and percentage of total DID the current account owns"
      >
        DID owned: {numDID} ({pctDID}%)
      </Menu.Item>
    )
  }
}

const mapStateToProps = state => ({
  numDID: state.user.user.numDID,
  totalSupplyDid: getTotalSupplyDID(state)
})

const mapDispatchToProps = dispatch => ({
  fetchTotalSupplyDID: () => dispatch(fetchTotalSupplyDID())
})

export default connect(mapStateToProps, mapDispatchToProps)(DIDOwnership)
