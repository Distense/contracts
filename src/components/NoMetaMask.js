import React, { Component } from 'react'
import { connect } from 'react-redux'
import { Grid, Message } from 'semantic-ui-react'

import { getConnectedCorrectNetwork, hasWeb3 } from '../features/user/reducers'
import { NETWORK_NAME } from '../features/user/network'

export class MetaMaskWarning extends Component {
  constructor(props) {
    super(props)
    this.state = {}
  }

  render() {
    const { hasWeb3, correctNetwork } = this.props

    return (
      <div>
        <Grid>
          <Grid.Column>
            <Grid.Row className="margin-bottom-rem">
              {(!hasWeb3 || !correctNetwork) && (
                <Message>
                  <Message.Item>
                    If you don't have it installed, please install{' '}
                    <a target="_blank" href="https://metamask.io/">
                      MetaMask
                    </a>{' '}
                    and connect to {NETWORK_NAME}
                  </Message.Item>
                </Message>
              )}
              {/*{hasWeb3 &&*/}
              {/*!correctNetwork && (*/}
              {/*<Message>*/}
              {/*<Message.Item>*/}
              {/*It seems like you have MetaMask installed but you are not*/}
              {/*connected to the right network*/}
              {/*</Message.Item>*/}
              {/*<Message.Item>*/}
              {/*Please connect to {NETWORK_NAME}*/}
              {/*</Message.Item>*/}
              {/*</Message>*/}
              {/*)}*/}
            </Grid.Row>
          </Grid.Column>
        </Grid>
      </div>
    )
  }
}

const mapStateToProps = state => ({
  hasWeb3: hasWeb3(state),
  correctNetwork: getConnectedCorrectNetwork(state)
})

export default connect(mapStateToProps, null)(MetaMaskWarning)
