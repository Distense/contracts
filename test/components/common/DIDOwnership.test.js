import React from 'react'
import { shallow } from 'enzyme'
import renderer from 'react-test-renderer'
import { MemoryRouter } from 'react-router-dom'
import { Provider } from 'react-redux'
import configureMockStore from 'redux-mock-store'
import { DIDOwnership } from '../../../src/components/DIDOwnership'

function setup(numDID = 0, totalSupplyDid = 0) {
  const props = {
    fetchTotalSupplyDID: jest.fn(),
    totalSupplyDid,
    numDID
  }
  const comp = shallow(<DIDOwnership {...props} />)

  return { comp, props }
}

describe('<DIDOwnership />', () => {
  it('renders without crashing', () => {
    const { comp } = setup()
    expect(comp).toBeDefined()
  })

  describe('calculates percentage owned correctly', () => {
    it('when percentage calculated is NaN it should return 0', () => {
      const { comp } = setup()
      expect(comp.instance().calculatePercentageOwned()).toEqual('0')
    })

    it('when a user owns zero DID it should return 0', () => {
      const { comp } = setup(0, 100)
      expect(comp.instance().calculatePercentageOwned()).toEqual('0')
    })

    it('when a user owns ALL of the DID it should return 100', () => {
      const { comp } = setup(100, 100)
      expect(comp.instance().calculatePercentageOwned()).toEqual('100')
    })

    it('when a user owns a percentage that calculates to an even tenth (54.1) it should render two decimal places (54.10)', () => {
      const { comp } = setup(25.2, 100)
      expect(comp.instance().calculatePercentageOwned()).toEqual('25.20')
    })
  })

  test('snapshot', () => {
    const { props } = setup()
    const mockstore = configureMockStore()
    const store = mockstore({})
    const tree = renderer
      .create(
        <Provider store={store}>
          <MemoryRouter>
            <DIDOwnership {...props} />
          </MemoryRouter>
        </Provider>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
