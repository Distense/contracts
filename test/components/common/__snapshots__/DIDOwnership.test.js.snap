// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DIDOwnership /> snapshot 1`] = `
<div
  className="item footer-item"
  onClick={[Function]}
  title="The number of and percentage of total DID the current account owns"
>
  DID owned: 
  0
   (
  0
  %)
</div>
`;
